### PR TITLE
Support reliability

### PR DESCRIPTION
**Pull request by winniewytse**

### CODE REVIEW

File: R/get_fscore.R [hunk 1]
```diff
@@ -129,6 +135,36 @@ get_fs_lavaan <- function(lavobj,
...
+                          length(attr(out, "fsb")) > 1)
+    if (multifactor) {
+      warning("Compution of reliability for a multi-factor model is not ",
+              "currently supported. ")
+    } else {
+      if (length(group) == 0) {
+        attr(out, "reliability") <- compute_rel(est, vcov(lavobj))
+      } else {
+        if (any(unlist(lapply(est, \(x) x$psi)) != 1)) {
+          warning("Computation of reliability for multiple groups is ",
```
Comment (marklhc) [inline R/get_fscore.R]: This should be for single group as well.

File: tests/testthat/test-get_fscore.R [hunk 2]
```diff
@@ -267,3 +267,22 @@ test_that("Correction factor is similar with single or multiple groups", {
   d2 <- fst2[[1]] - fst2[[2]]
   expect_lt(mean(abs(d2)), mean(abs(d1)))
 })
+
+########## Computing reliability ##########
+
+test_that("Regression factor scores", {
+  fs <- get_fs(PoliticalDemocracy[c("x1", "x2", "x3")],
+               corrected_fsT = TRUE, reliability = TRUE)
```
Comment (marklhc) [inline tests/testthat/test-get_fscore.R]: Currently the test does not use `std.lv = TRUE`

File: vignettes/reliability.Rmd [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+  \(x) get_a(x[1:3], diag(x[4:6])),
+  x = c(lam, diag(th))
+)
+# Estimated sampling variance of weights
+va <- jac_a %*% vcov(cfa_fit)[1:6, 1:6] %*% t(jac_a)
+aa <- tcrossprod(t(get_a(lam, th))) + va
+sum(diag(tcrossprod(lam) %*% aa)) /
+  sum(diag(cfa_fit@implied$cov[[1]] %*% aa))
+# Or using `get_fs_lavaan` (need to be fixed)
+R2spa::get_fs_lavaan(cfa_fit, method = "Bartlett", reliability = TRUE) |>
```
Comment (marklhc) [inline vignettes/reliability.Rmd:208]: Right now the function doesn't seem to compute the right reliability with Bartlett scores.

Comment (marklhc) [reply vignettes/reliability.Rmd:208]: Need to add a unit test as well.

**Comment by marklhc** [comment]
Thanks Wiinnie. I've made some changes since there are already some internal functions for computing the weight matrix. It should also work for non-diagonal $\Theta$ matrix too, but I haven't fully tested that. Let me know if you have any thoughts and questions.

File: R/get_fscore.R [hunk 4]
```diff
@@ -184,38 +184,21 @@ get_fs_lavaan <- function(lavobj,
...
-            group_n <- lavInspect(lavobj, what = "norig")
-            rels[g + 1] <- sum(rels * group_n / sum(group_n))
-            attr(out, "reliability") <-
-              setNames(as.list(rels), c(lavobj@Data@group.label, "overall"))
-          } else {
-            warning("Only diagonal theta matrix is currently supported. ")
-            attr(out, "reliability") <- NULL
-          }
-        }
+        is_std.lv = all(unlist(lapply(est, function(x) x$psi)) == 1)
```
Comment (winniewytse) [inline R/get_fscore.R:190]: Looks like this line of code can be revmoed

**Comment by winniewytse** [comment]
@marklhc Thanks for updating the `compute_fsrel()` function! The changes look great to me, and it's great that the function handles also non-diagonal Theta!

**Pull request merged**